test/guardian/permissions/bitwise_test.exs

```diff
@@ -224,7 +224,7 @@ defmodule Guardian.Permissions.BitwiseTest do
       conn = Guardian.Permissions.Bitwise.call(ctx.conn, opts)
 
       assert {403, _headers, body} = sent_resp(conn)
-      assert body == "{:unauthorized, :unauthorized}"
+      assert body == "{:unauthorized, \"Insufficient permission\"}"
```

Comment:
Don't we want to return just "Insufficient permission" here?

Reply:
@doomspork welcome to feedbacks

Reply:
I am not sure what would people prefer to get here 🤷‍♂️

Reply:
The including the tuple seems unnecessary don't you think? Tuples nor atoms are valid JSON so this isn't anything, right? It's not real sentence and it's not valid JSON.

Reply:
> The including the tuple seems unnecessary don't you think? Tuples nor atoms are valid JSON so this isn't anything, right?

Didn't write this so I am not sure, never used `bitwise` module before and the reason I asked for help.

But from what I understand this is being passed to the `auth_error` callback where you decide to respond with a valid JSON, text, or whatever you decide to do.

Reply:
I used it when I was debugging a bitwise error. What I thought could be cool was to write the input permission and the required permission, but maybe this is a little bit out of scope for now?

Reply:
@yordis I understand but we should improve the code anytime we make changes to it 😁In addition to testing the code, the tests can/should service has a type of documentation. Never would you really return a tuple like this, so let's change it, even if you didn't write it.

If you come to Guardian to either use the library or contribute, find your way to these tests, and see them returning something that _you'd never see in an application_ it could be a source of confusion — hell it's confused me and I'm one of the maintainers 😆

Let's update the test handler to return something like valid JSON so these tests can look remotely like they would in a real scenario.

Does that sounds good?

```elixir
  defmodule Handler do
    @moduledoc false

    import Plug.Conn
    @behaviour Guardian.Plug.ErrorHandler

    @impl Guardian.Plug.ErrorHandler
    def auth_error(conn, {type, reason}, _opts) do
      send_resp(conn, 403, Poison.encode!(%{type => reason}))
    end
  end```

Reply:
@doomspork since users will handle what to do with the error tuple anyway I will put as many details as possible on what happened (like @Hanspagh wants it), in any case, the user could decide if they want to send a generic error or not (this is why I put a generic error, I wasn't sure about it).

Now, if that is the case, maybe instead of returning a string back I will return an atom since the user can always decide to convert the atom to the proper string message.

Or triple tuple with the last value being the human-readable message (I would simply work with atoms).

Reply:
@yordis I'm only talking about updating this test. I'm not sure I follow all these other changes?

Reply:
https://github.com/ueberauth/guardian/blob/77c7acdba384563160b935bf75d233bf71f495b7/test/guardian/permissions/bitwise_test.exs#L27